### PR TITLE
refactor(site): improve minor queries and visuals on external auth

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
- Minor visual improvements
- Minor query refactoring

PS: To really improve the design of this, I would need to refactor the settings layout a bit which is out of the scope for now.

Before: 
Screenshot 2023-12-06 at 13 39 57

Now:
Screenshot 2023-12-06 at 13 40 05

### CODE REVIEW

**Comment by BrunoQuaresma** [comment]
@kylecarbs @Emyrk just marking you to share a little bit of how we are organizing the queries and mutations

**Review by Emyrk — COMMENTED** [review]
The table spacing is a lot better :+1:

**Review by buenos-nachos — APPROVED** [review]
Looks good!

Had one concern about the `onSuccess` callback, but I'll leave it to you to decide if that's worth addressing now, or when we make the migration to RQ v5

File: site/src/api/queries/externalAuth.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+) => {
+  return {
+    queryFn: () =>
+      API.exchangeExternalAuthDevice(providerId, {
+        device_code: deviceCode,
+      }),
+    queryKey: ["external-auth", providerId, "device", deviceCode],
+    onSuccess: async () => {
+      // Force a refresh of the Git auth status.
+      await queryClient.invalidateQueries(["external-auth", providerId]);
```
[the remark this answers is not in the document]
Comment (BrunoQuaresma) [reply site/src/api/queries/externalAuth.ts:40]: That is absolutely true! Let me remove it.

[the remark this answers is not in the document]
Comment (BrunoQuaresma) [reply site/src/api/queries/externalAuth.ts:40]: After taking a look into this specific query, I think it is not trivial to remove it right now but we definitely can get back to this when migrating to v5.

**Pull request merged**